### PR TITLE
Update Poco to 1.9.3.

### PR DESCRIPTION
**Pull request by akuzm**
This is required for MongoDB URI support.

Category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry:
Update Poco to 1.9.3 in preparation for MongoDB URI support.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alesapin** [comment]
Clone error:
```
No submodule mapping found in .gitmodules for path 'gradle'
Failed to recurse into submodule path 'contrib/poco'
```

**Comment by akuzm** [comment]
Should update to 1.9.4: https://pocoproject.org/blog/?p=1099

File: libs/libdaemon/src/BaseDaemon.cpp [hunk 1]
```diff
@@ -148,6 +148,7 @@ class SignalListener : public Poco::Runnable
         {
             int sig = 0;
             DB::readBinary(sig, in);
+            LOG_TRACE(log, "Received signal " << sig);
```
Comment (alexey-milovidov) [inline libs/libdaemon/src/BaseDaemon.cpp]: 1. strsignal.
2. We already print it somewhere.

**Comment by filimonov** [comment]
Can you also look on that while we are here?
https://github.com/ClickHouse/ClickHouse/issues/7736#issuecomment-552922270

**Comment by akuzm** [comment]
> Can you also look on that while we are here?
> [#7736 (comment)](https://github.com/ClickHouse/ClickHouse/issues/7736#issuecomment-552922270)

Not until I finish this update, and my perspective on this is quite grim, given the Mystery Integration Test Failure I'm seeing in CI but not locally.

**Comment by alexey-milovidov** [comment]
Tsan report is relevant and it looks familiar to me.
Did you integrate all the patches?

**Comment by alexey-milovidov** [comment]
It looks almost ready, let's finish.

**Comment by alexey-milovidov** [comment]
`00834_kill_mutation` - flacky test.

`test_dictionaries_update_and_reload/test.py::test_reload_after_loading` - it may be attributed to https, let's check carefully.

**Comment by akuzm** [comment]
Split build failed because of CI error -- works locally.

**Comment by alexey-milovidov** [comment]
Split build smoke test: #10820

**Pull request merged**